pennylane/qchem/tapering.py

```diff
@@ -290,28 +309,33 @@ def taper(h, generators, paulixops, paulix_sector):
 
     # cast to pauli sentence
     ps_u = pauli_sentence(u)
-    ps_h = pauli_sentence(h)
 
-    h = (ps_u * ps_h * ps_u).hamiltonian()  # cast back to hamiltonian
+    ps_h = h
+    if not isinstance(ps_h, qml.pauli.PauliSentence):
+        ps_h = pauli_sentence(h)
 
-    val = np.ones(len(h.terms()[0])) * complex(1.0)
+    ts_h = qml.pauli.PauliSentence()
+    for ps in _split_pauli_sentence(ps_h, max_size=PAULI_SENTENCE_MEMORY_SPLITTING_SIZE):
+        ts_h += ps_u * ps * ps_u  # helps restrict the peak memory usage for u @ h @ u
+    h = ts_h.hamiltonian()  # cast back to hamiltonian
 
     wireset = u.wires + h.wires
     wiremap = dict(zip(wireset, range(len(wireset) + 1)))
     paulix_wires = [x.wires[0] for x in paulixops]
 
-    for idx, w in enumerate(paulix_wires):
-        for i in range(len(h.terms()[0])):
-            s = qml.pauli.pauli_word_to_string(h.terms()[1][i], wire_map=wiremap)
-            if s[w] == "X":
-                val[i] *= paulix_sector[idx]
-
     o = []
+    h_coeffs, h_ops = h.terms()
+    val = np.ones(len(h_coeffs))
+
     wires_tap = [i for i in h.wires if i not in paulix_wires]
     wiremap_tap = dict(zip(wires_tap, range(len(wires_tap) + 1)))
 
-    for i in range(len(h.terms()[0])):
-        s = qml.pauli.pauli_word_to_string(h.terms()[1][i], wire_map=wiremap)
+    for i in range(len(h_coeffs)):
+        s = qml.pauli.pauli_word_to_string(h_ops[i], wire_map=wiremap)
+
+        for idx, w in enumerate(paulix_wires):
+            if s[w] == "X":
+                val[i] *= paulix_sector[idx]
```

Comment:
```suggestion
    for i, term in enumerate(h_ops):
        term_string = qml.pauli.pauli_word_to_string(term, wire_map=wiremap)

        for w, sector in zip(paulix_wires, paulix_sector):
            if term_string[w] == "X":
                val[i] *= sector
```
Or potentially something like:
```
tapered_pauli_sentence = {}

for term in ts_h:
    tapered_pauli_sentence[term] = 1.0
    for w, sector in zip(paulix_wires, paulix_sector):
        if term[w] == "X":
           tapered_pauli_sentence[term] *= sector
```
Pretty sure that's the same thing.  This way you could avoid going back and forth between representations completely.